test/unit/pages/patientdata.test.js

```diff
@@ -132,87 +138,81 @@ describe('PatientData', function () {
-    var elem = TestUtils.renderIntoDocument(<PatientData {...props}/>);
-    expect(elem).to.be.ok;
-    
-    var x = TestUtils.findRenderedDOMComponentWithClass(elem, 'patient-data-message');
-    expect(x).to.be.ok;
-  });
+      // Try out using the spread props syntax in JSX
+      var elem = TestUtils.renderIntoDocument(<PatientData {...props}/>);
+      expect(elem).to.be.ok;
+      
+      var x = TestUtils.findRenderedDOMComponentWithClass(elem, 'patient-data-message');
+      expect(x).to.be.ok;
+    });
 
-  it ('should render when data is present for current patient', function() {
-    var props = {
-      timePrefs: {
-        timezoneAware: false,
-        timezoneName: null
-      },
-      patient: {
-        userid: 40
-      },
-      patientData: {
-        40: { data: [ 1, 2] }
-      },
-      fetchingPatient: false,
-      fetchingPatientData: false,
-      queryParams: {},
-      trackMetric: sinon.stub()
-    };
-
-    PatientData.__set__('Modal', React.createClass({
-      render: function() {
-        console.log('Mock', this.props);
-        return <div className='fake-modal-view'/>;
-      }
-    }));
-
-    // Try out using the spread props syntax in JSX
-    var elem = TestUtils.renderIntoDocument(<PatientData {...props}/>);
-    expect(elem).to.be.ok;
-    
-    // var x = TestUtils.findRenderedDOMComponentWithClass(elem, 'fake-modal-view');
-    // expect(x).to.be.ok;
+    it ('should render the no data message when no data is present for current patient', function() {
+      var props = {
+        timePrefs: {
+          timezoneAware: false,
+          timezoneName: null
+        },
+        patient: {
+          userid: 40
+        },
+        patientData: {
+          41: { data: [] }
+        },
+        fetchingPatient: false,
+        fetchingPatientData: false,
+        queryParams: {},
+        trackMetric: sinon.stub()
+      };
+
+      // Try out using the spread props syntax in JSX
+      var elem = TestUtils.renderIntoDocument(<PatientData {...props}/>);
+      expect(elem).to.be.ok;
+      
+      var x = TestUtils.findRenderedDOMComponentWithClass(elem, 'patient-data-message');
+      expect(x).to.be.ok;
+    });
+
+    it ('should render when data is present for current patient', function() {
+      var props = {
+        timePrefs: {
+          timezoneAware: false,
+          timezoneName: null
+        },
+        patient: {
+          userid: 40
+        },
+        patientData: {
+          40: { data: [ 1, 2] }
+        },
+        fetchingPatient: false,
+        fetchingPatientData: false,
+        queryParams: {},
+        trackMetric: sinon.stub()
+      };
+
+      var pdElem = React.createElement(PatientData, props);
+      var elem = TestUtils.renderIntoDocument(pdElem);
+      expect(elem).to.be.ok;
+      
+      // var x = TestUtils.scryRenderedDOMComponentsWithClass(elem, 'fake-modal-view');
+      // console.log(x.length);
+      //expect(x).to.be.ok;
```

Comment:
It really baffled me! I tried a number of different tools but to no avail.
I finally found blog on rewire.js and thought, considering the same tools
it would all be fine, but sadly not :(

Look forward to bringing our minds together on this one!

On Mon, Jun 22, 2015 at 6:44 AM, Jana Beck <notifications@github.com> wrote:

> In test/unit/pages/patientdata.test.js
> <https://github.com/tidepool-org/blip/pull/239#discussion_r32906487>:
>
> > +        patientData: {
> > +          40: { data: [ 1, 2] }
> > +        },
> > +        fetchingPatient: false,
> > +        fetchingPatientData: false,
> > +        queryParams: {},
> > +        trackMetric: sinon.stub()
> > +      };
> > +
> > +      var pdElem = React.createElement(PatientData, props);
> > +      var elem = TestUtils.renderIntoDocument(pdElem);
> > +      expect(elem).to.be.ok;
> > +
> > +      // var x = TestUtils.scryRenderedDOMComponentsWithClass(elem, 'fake-modal-view');
> > +      // console.log(x.length);
> > +      //expect(x).to.be.ok;
>
> Yeah, I'll definitely take a closer look soon. This is one of the things I
> struggled with recently too when trying to write Flux tests without Jest.
> Hopefully between the two of us we can figure something out!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tidepool-org/blip/pull/239/files#r32906487>.
>